core/lib/vm/src/tracers/default_tracers.rs

```diff
@@ -222,37 +206,64 @@ impl<S: WriteStorage, H: HistoryMode> DefaultExecutionTracer<S, H> {
             .populate_page(BOOTLOADER_HEAP_PAGE as usize, memory, current_timestamp);
         self.final_batch_info_requested = false;
     }
+
+    fn should_stop_execution(&self) -> TracerExecutionStatus {
+        match self.execution_mode {
+            VmExecutionMode::OneTx => {
+                if self.tx_has_been_processed() {
+                    return TracerExecutionStatus::Stop(TracerExecutionStopReason::Finish);
+                }
+            }
+            VmExecutionMode::Bootloader => {
+                if self.ret_from_the_bootloader == Some(RetOpcode::Ok) {
+                    return TracerExecutionStatus::Stop(TracerExecutionStopReason::Finish);
+                }
+            }
+            VmExecutionMode::Batch => {}
+        };
+        if self.validation_run_out_of_gas() {
+            return TracerExecutionStatus::Stop(TracerExecutionStopReason::Abort(
+                Halt::ValidationOutOfGas,
+            ));
+        }
+        TracerExecutionStatus::Continue
```

Comment:
this is a little hard to follow - we do `match`, don't return only in the `Batch` variabt, and then do an `if`. I think it would be cleaner with

```
match self.execution_mode {
            VmExecutionMode::OneTx if self.tx_has_been_processed() => 
                      TracerExecutionStatus::Stop(TracerExecutionStopReason::Finish)
            }
            VmExecutionMode::Bootloader if self.ret_from_the_bootloader == Some(RetOpcode::Ok) => 
                    TracerExecutionStatus::Stop(TracerExecutionStopReason::Finish);
            }
            VmExecutionMode::Batch if self.validation_run_out_of_gas() => 
TracerExecutionStatus::Stop(TracerExecutionStopReason::Abort(
                Halt::ValidationOutOfGas,
        _ => 
        TracerExecutionStatus::Continue
            
        };
```

Reply:
The two variants can be squashed into one i think with `|`

Reply:
We have to always check `self.validation_run_out_of_gas`. Independent from execution_mode. 
Thanks for the other suggestions :) 


> The two variants can be squashed into one i think with |

Only without if..

Reply:
Why? We dobn't check it now since we return